pyxem/hyperspy_extension.yaml

```diff
@@ -18,109 +18,115 @@
 
 signals:
   Diffraction1D:
-    signal_type: "Diffraction1D"
+    signal_type: diffraction1d
     signal_dimension: 1
     dtype: real
     lazy: False
     module: pyxem.signals.diffraction1d
   Diffraction2D:
-    signal_type: "Diffraction2D"
+    signal_type: diffraction2d
     signal_dimension: 2
     dtype: real
     lazy: False
     module: pyxem.signals.diffraction2d
   LazyDiffraction1D:
-    signal_type: "LazyDiffraction1D"
+    signal_type: diffraction1d
     signal_dimension: 1
     dtype: real
     lazy: True
     module: pyxem.signals.diffraction1d
   LazyDiffraction2D:
-    signal_type: "LazyDiffraction2D"
+    signal_type: diffraction2d
     signal_dimension: 2
     dtype: real
     lazy: True
     module: pyxem.signals.diffraction2d
   ElectronDiffraction1D:
-    signal_type: "ElectronDiffraction1D"
+    signal_type: electron_diffraction1d
     signal_dimension: 1
     dtype: real
     lazy: False
     module: pyxem.signals.electron_diffraction1d
   ElectronDiffraction2D:
-    signal_type: "ElectronDiffraction2D"
+    signal_type: electron_diffraction2d
     signal_dimension: 2
     dtype: real
     lazy: False
     module: pyxem.signals.electron_diffraction2d
   LazyElectronDiffraction1D:
-    signal_type: "LazyElectronDiffraction1D"
+    signal_type: electron_diffraction1d
     signal_dimension: 1
     dtype: real
     lazy: True
     module: pyxem.signals.electron_diffraction1d
   LazyElectronDiffraction2D:
-    signal_type: "LazyElectronDiffraction2D"
+    signal_type: electron_diffraction2d
     signal_dimension: 2
     dtype: real
     lazy: True
     module: pyxem.signals.electron_diffraction2d
   DiffractionVectors:
-    signal_type: "DiffractionVectors"
-    signal_dimension: None
+    signal_type: diffraction_vectors
+    signal_dimension: 0
+    dtype: real
+    lazy: False
+    module: pyxem.signals.diffraction_vectors
+  DiffractionVectors2D:
+    signal_type: diffraction_vectors
+    signal_dimension: 2
     dtype: real
     lazy: False
     module: pyxem.signals.diffraction_vectors
   TemplateMatchingResults:
-    signal_type: "TemplateMatchingResults"
-    signal_dimension: None
+    signal_type: template_matching
+    signal_dimension: 2
     dtype: real
     lazy: False
     module: pyxem.signals.indexation_results
   VectorMatchingResults:
-    signal_type: "VectorMatchingResults"
-    signal_dimension: None
+    signal_type: vector_matching
+    signal_dimension: 0
     dtype: real
     lazy: False
     module: pyxem.signals.indexation_results
   CrystallographicMap:
-    signal_type: "CrystallographicMap"
-    signal_dimension: None
+    signal_type: crystallographic_map
+    signal_dimension: 1
     dtype: real
     lazy: False
     module: pyxem.signals.crystallographic_map
   DisplacementGradientMap:
-    signal_type: "DisplacementGradientMap"
+    signal_type: tensor_field
```

Comment:
Are we happy with this signal_type name (I know you're not introducing it in this PR, but seems as good a time as any to review it) - would "displacement_gradient_map" be better to be explicit?

Reply:
I don't know what would be better and I don't have any view on it, so I have changed as you suggested.